cmd/zsysd/completion.go

```diff
@@ -11,18 +12,66 @@ import (
 func installCompletionCmd(rootCmd *cobra.Command) {
 	prog := rootCmd.Name()
 	var completionCmd = &cobra.Command{
-		Use:   "completion",
-		Short: i18n.G("Generates bash completion scripts"),
-		Long: fmt.Sprintf(i18n.G(`To load completion run
+		Use:   "completion [bash|zsh|powershell]",
+		Short: i18n.G("Generates completion scripts"),
+		Long: strings.ReplaceAll(i18n.G(`To load completions:
 
-. <(%s completion)
+Bash:
 
-To configure your bash shell to load completions for each session add to your ~/.bashrc or ~/.profile:
+  $ source <(%s completion bash)
 
-. <(%s completion)
-`), prog, prog),
+  # To load completions for each session, execute once:
+  # Linux:
+  $ %s completion bash > /etc/bash_completion.d/%s
+  # macOS:
+  $ %s completion bash > /usr/local/etc/bash_completion.d/%s
+
+Zsh:
+
+  # If shell completion is not already enabled in your environment,
+  # you will need to enable it.  You can execute the following once:
+
+  $ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+  # To load completions for each session, execute once:
+  $ %s completion zsh > "${fpath[1]}/_%s"
+
+  # You will need to start a new shell for this setup to take effect.
+
+PowerShell:
+
+  PS> %s completion powershell | Out-String | Invoke-Expression
+
+  # To load completions for every new session, run:
+  PS> %s completion powershell > %s.ps1
+  # and source this file from your PowerShell profile.
+`), "%s", prog),
+		ValidArgs: []string{"bash", "zsh", "powershell"},
+		Args:      cobra.MaximumNArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
-			genBashCompletion(rootCmd, os.Stdout)
+			tmpShell := os.Getenv("SHELL")
+			shell := "unknown"
```

Comment:
nitpick: I would just here declare and simplify the whole stanza here:
```
var shell string
if len(args) > 0 && args[0] != "" {
	shell = args[0]
} 
shell = filepath.Base(shell)

switch shell…
```

Then, the error message will only print "Shell preset unknown: <value of shell>", I would add something like: "Please specify between bash, zsh, powershell".

Reply:
Cleaned up, as for the error message, it is followed by the usage output.
Do you still think this should be changed?